Vagrant provisioning updates

Matt-Yorkley opened the pull request:
I've made a few adjustments to get Vagrant deploys working smoothly. I've just been doing this over the weekend as a learning exercise for working with Ansible and Vagrant. 

There were a lot of errors and some package issues with the previous version, but now it all runs smoothly in ~20 minutes with:

`$ vagrant up`
`$ ansible-playbook site.yml -i inventory/dev --limit=vagrant`

![screenshot from 2018-06-11 17-32-50](https://user-images.githubusercontent.com/9029026/41245964-78b81e8a-6da1-11e8-841c-d6b410cd0a54.png)

I'd appreciate any feedback/comments. If we merge this we can also update the Vagrant wiki page in ofn-install, with: "Tested with Vagrant 2.1.1 and Virtualbox 5.2.10" and the updated command to provision it properly (it needs `--limit=vagrant` and doesn't need `-u vagrant`).

sauloperez commented on roles/unicorn_user/tasks/main.yml:
```diff
@@ -16,6 +16,17 @@
+- name: add ssh key alternative
+  authorized_key:
+    user: "{{ unicorn_user }}"
+    key: "{{ lookup('file', lookup('env','HOME') + ssh_key_path) }}"
```
It would be a bit more obvious for the user if `ssh_key_path` behaved as an actual path allowing `ssh_key_path=~/.ssh/id_rsa.pub` and also more flexible; it would allow public keys from any location.

So I suggest this to be

```sh
key: "{{ ssh_key_path }}"
```

Matt-Yorkley replied:
I copied this from the other ssh key setup for the default user in https://github.com/openfoodfoundation/ofn-install/blob/530741c5d82f6c47b0e196666b0b13920310fedf/roles/default_user/tasks/main.yml#L20

Is it worth changing that one as well?

Matt-Yorkley replied:
It looks like they have to match for it to run, so I guess yes is the answer...

Matt-Yorkley replied:
On testing it, it looks like it won't accept paths that start with `~/`, which I guess is why the same method of using `{{ lookup('file', lookup('env','HOME') + ssh_key_path) }}` was used previously in /roles/default_user/tasks/main.yml.

sauloperez replied:
Ah ok, nevermind then... Although I find it really weird...

Matt-Yorkley replied:
The answer was somewhere in between: `{{ lookup('file', ssh_key_path) }}` can handle the tilde in the filepath, the previous addition of `lookup('env','HOME')` wasn't needed.

Matt-Yorkley commented:
@sauloperez @enricostano

mkllnk commented on Vagrantfile:
```diff
@@ -10,13 +16,11 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
-    # The default catalogue moved and this image is not found on
-    # atlas.hashicorp.com any more. Older versions of vagrant still try
-    # to access the old server and need the URL explicitely.
-    config.vm.box_url = "https://app.vagrantup.com/ubuntu/trusty64"
```
Okay, scrap that. My backported vagrant knows the new URLs.

mkllnk commented:
What do you think about splitting `inventory/dev` into `inventory/vagrant` and `inventory/lxc`? I don't think that anyone of us wants to run both at the same time.

mkllnk commented:
Yay, I just ran this and it worked beautifully. Thank you. I would just like to know about the Ubuntu image.

Matt-Yorkley commented:
Nice catch @mkllnk, after investigating a bit more I can see there's an unaddressed issues here which could interfere with provisioning new servers. It's related to this: https://github.com/openfoodfoundation/ofn-install/issues/112

I'll fix that issue first then update the vagrant file to use the official xenial64 image. :+1:

Matt-Yorkley commented:
Should we scrap `lxc`? I think it's outdated/unmaintained...

Matt-Yorkley commented:
I've fixed the xenial/python issue in #196. Once that's merged we can use this PR which is now set to use the official xenial64 image. I've tested the both together with vagrant using the official image and it's working fine.

mkllnk commented:
Oh, I just got this error:
```
fatal: [local_vagrant]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'file'. Error was a <class 'ansible.errors.AnsibleError'>, original message: could not locate file in lookup: /home/maikel~/.ssh/id_rsa.pub"}
```

That file lookup doesn't seem to work properly.

mkllnk commented:
Ah, no, forget that. I didn't have this merged. :-)
